### ISSUE TITLE
UnicodeDecodeError on GET /mwoauth/initiate with non-english mediawiki

### ISSUE THREAD

**Issue by xunto** [mediawiki-utilities/python-mwoauth#33]
Versions:
```
mwoauth==0.3.3
Flask==1.0.2
requests==2.22.0
requests-oauthlib==1.2.0
```

Upon requesting /mwoauth/initiate I got this:
```
127.0.0.1 - - [07/Aug/2019 14:46:07] "GET /mwoauth/initiate/?next=mwoauth.mwoauth_identify HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 2309, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 2295, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/lib/python3.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 1741, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3.7/site-packages/flask_cors/extension.py", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/lib/python3.7/site-packages/mwoauth/flask.py", line 107, in mwoauth_initiate
    mw_authorizer_url, request_token = self._handshaker().initiate()
  File "/usr/lib/python3.7/site-packages/mwoauth/handshaker.py", line 89, in initiate
    user_agent=self.user_agent)
  File "/usr/lib/python3.7/site-packages/mwoauth/functions.py", line 92, in initiate
    credentials = parse_qs(r.content)
  File "/usr/lib/python3.7/urllib/parse.py", line 675, in parse_qs
    max_num_fields=max_num_fields)
  File "/usr/lib/python3.7/urllib/parse.py", line 710, in parse_qsl
    qs, _coerce_result = _coerce_args(qs)
  File "/usr/lib/python3.7/urllib/parse.py", line 123, in _coerce_args
    return _decode_args(args) + (_encode_result,)
  File "/usr/lib/python3.7/urllib/parse.py", line 107, in _decode_args
    return tuple(x.decode(encoding, errors) if x else '' for x in args)
  File "/usr/lib/python3.7/urllib/parse.py", line 107, in <genexpr>
    return tuple(x.decode(encoding, errors) if x else '' for x in args)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd0 in position 7: ordinal not in range(128)
```

After debugging I found that it happens here:
https://github.com/mediawiki-utilities/python-mwoauth/blob/464030a314a524a053a52812746ecd0eaaec02b4/mwoauth/functions.py#L92

with following response content:
```
b'Error: \xd0\x9f\xd1\x80\xd0\xbe\xd0\xb8\xd0\xb7\xd0\xbe\xd1\x88\xd0\xbb\xd0\xb0 \xd0\xbe\xd1\x88\xd0\xb8\xd0\xb1\xd0\xba\xd0\xb0 \xd0\xb2 \xd0\xbf\xd1\x80\xd0\xbe\xd1\x82\xd0\xbe\xd0\xba\xd0\xbe\xd0\xbb\xd0\xb5 OAuth: Invalid consumer key'
```

Which translates:
```
Error: Произошла ошибка в протоколе OAuth: Invalid consumer key
```

**Comment by halfak** [comment]
Curses, mediawiki.  Why do you always have to switch up your response types!?  It looks like we should package this error and report it to the user.  

Normally, we would handle surprising results but it looks like UTF-8 decoding here is necessary.

**Comment by halfak** [comment]
This should be addressed in mwoauth version 0.3.5.  I've uploaded this version to pypi.
